client.go

```diff
@@ -29,6 +68,92 @@ type RestClient struct {
 	baseUrl url.URL
 }
 
+func NewNetworkClient(config NetworkConfig) (client *Client, err error) {
+	var apiUrl *url.URL = nil
+
+	switch {
+	case config.api == nil && config.network == nil:
+		err = errors.New("aptos api url or network is required")
+		return
+	case config.api != nil:
+		apiUrl, err = url.Parse(*config.api)
+		if err != nil {
+			return
+		}
+	case *config.network == localnet:
+		apiUrl, err = url.Parse(localnet_api)
+		if err != nil {
+			return
+		}
+	case *config.network == devnet:
+		apiUrl, err = url.Parse(devnet_api)
+		if err != nil {
+			return
+		}
+	case *config.network == testnet:
+		apiUrl, err = url.Parse(testnet_api)
+		if err != nil {
+			return
+		}
+	case *config.network == mainnet:
+		apiUrl, err = url.Parse(mainnet_api)
+		if err != nil {
+			return
+		}
+	default:
+		err = errors.New("network name is unknown, please put localnet, devnet, testnet, or mainnet")
+		return
+	}
+	var faucetUrl *url.URL = nil
+
+	switch {
+	case config.faucet == nil && config.network == nil:
+		err = errors.New("aptos faucet url or network is required")
+		return
+	case config.faucet != nil:
+		faucetUrl, err = url.Parse(*config.faucet)
+		if err != nil {
+			return
+		}
+	case *config.network == localnet:
+		faucetUrl, err = url.Parse(localnet_faucet)
+		if err != nil {
+			return
+		}
+	case *config.network == devnet:
+		faucetUrl, err = url.Parse(devnet_faucet)
+		if err != nil {
+			return
+		}
+	case *config.network == testnet:
+		faucetUrl, err = url.Parse(testnet_faucet)
+		if err != nil {
+			return
+		}
+	case *config.network == mainnet:
+		faucetUrl = nil
+	default:
+		err = errors.New("network name is unknown, please put localnet, devnet, testnet, or mainnet")
+		return
+	}
+
+	// TODO: add indexer
+
+	restClient := new(RestClient)
+	restClient.baseUrl = *apiUrl
+	restClient.client.Timeout = 60 * time.Second // TODO: Make configurable
+	faucetClient := &FaucetClient{
+		restClient,
+		faucetUrl,
+	}
+	client = &Client{
+		*restClient,
+		*faucetClient,
+	}
+	return
+}
+
+// TODO: Deprecate
```

Comment:
just do it, only cmd/goclient/goclient.go to fix.

Reply:
Will do.